### ISSUE TITLE
Listing products (attributes) fails when SQL mode `only_full_group_by` is in effect

### ISSUE THREAD

**Issue by rimas-kudelis** [BitBagCommerce/SyliusElasticsearchPlugin#114]
I can't find a bug about this, so creating a new one. When SQL mode `only_full_group_by` is in effect (which is the default on latest MySQL versions), attempting to load product index results in an error:

![paveikslas](https://user-images.githubusercontent.com/2812673/81180536-d6918380-8fb3-11ea-949f-7161d311bd19.png)

Note that I'm using a somewhat forked version of your plugin, so there is a slight possibility that this is caused by my changes. However, I'm quite certain that it isn't.

**Comment by bondie** [comment]
You can set database options like this:

```yaml
# config/packages/doctrine.yaml
doctrine:
    dbal:
        options:
            1002: 'SET sql_mode=(SELECT REPLACE(@@sql_mode, "ONLY_FULL_GROUP_BY", ""))'
```

**Comment by rimas-kudelis** [comment]
I understand that, but your suggestion is just a workaround which masks the issue, not a real fix.

I've fixed it a bit differently in my copy:
- I've changed the `return` statement in [ProductAttributeValueRepository.php#L32-L38](https://github.com/BitBagCommerce/SyliusElasticsearchPlugin/blob/741834b039425735f153825d5fbb264b311ec8db/src/Repository/ProductAttributeValueRepository.php#L32-L38) to read as follows:
```php
        return $queryBuilder
            ->select('o.localeCode, o.'.$storageType.' as value')
            ->where('o.attribute = :attribute')
            ->groupBy('o.' . $storageType)
            ->addGroupBy('o.localeCode')
            ->setParameter(':attribute', $productAttribute)
            ->getQuery()
            ->getResult()
        ;
```

And changed [lines 62-72 in ProductAttributesMapper.php](https://github.com/BitBagCommerce/SyliusElasticsearchPlugin/blob/741834b039425735f153825d5fbb264b311ec8db/src/Form/Type/ChoiceMapper/ProductAttributesMapper.php#L62-L71) to read as follows:
```php
            if ($productAttributeValue instanceof ProductAttributeValueInterface) {
                $value = $productAttributeValue->getValue();
            } else {
                $value = $productAttributeValue['value'];
            }

            $configuration = $productAttribute->getConfiguration();
```

The first change causes only relevant fields to be selected and uses them both for `GROUP BY`, but the result is now returned as an array of arrays instead of an array of objects. The second change accounts for that possibility.

Even ProductAttributeValueRepositoryInterface did not need changing, since it only typehints the return value is an array.
